### PR TITLE
[PO-10] Add Transactions generator

### PR DESCRIPTION
**Pull request by mirkoAlic**
The tests for the Transactions generator could fail if the random transaction has the same sender and receipt, this is because of an error in the stake distribution implementation.

### CODE REVIEW

File: src/main/scala/io/iohk/praos/util/TransactionsGenerator.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package io.iohk.praos.util
+
+import scala.annotation.tailrec
+import scala.util.Random
+import io.iohk.praos.crypto.Key
+import io.iohk.praos.domain.{Stake, StakeDistribution, Transaction, applyTransaction}
+
+object TransactionsGenerator {
+
+  type Stakeholder = (Key, Stake)
```
Comment (ntallar) [inline src/main/scala/io/iohk/praos/util/TransactionsGenerator.scala]: Minor detail: Maybe this type alias should be `private` to differentiate from the `case class StakeHolder`?

File: src/main/scala/io/iohk/praos/util/TransactionsGenerator.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      senderPublicKey = sender._1,
+      recipientPublicKey = recipient._1,
+      stake = chooseRandomStakeFrom(stakeUpperBound = sender._2)
+    )
+  }
+
+  def generateTxs(stakeDistribution: StakeDistribution, n: Int): List[Transaction] = {
+    @tailrec
+    def rec(stakeDistribution: StakeDistribution, n: Int, acc: List[Transaction]): List[Transaction] = {
+      val transaction = generateTx(stakeDistribution)
```
Comment (ntallar) [inline src/main/scala/io/iohk/praos/util/TransactionsGenerator.scala]: Minor detail: Maybe this transaction should only be created when the `n` is greater than 0?

File: src/main/scala/io/iohk/praos/util/TransactionsGenerator.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    val sender = chooseRandomStakeholderFrom(stakeDistribution)
+    val recipient = chooseRandomStakeholderFrom(stakeDistribution)
+    Transaction(
+      senderPublicKey = sender._1,
+      recipientPublicKey = recipient._1,
+      stake = chooseRandomStakeFrom(stakeUpperBound = sender._2)
+    )
+  }
+
+  def generateTxs(stakeDistribution: StakeDistribution, n: Int): List[Transaction] = {
```
Comment (ntallar) [inline src/main/scala/io/iohk/praos/util/TransactionsGenerator.scala:36]: Should we add a `require(n>=0)`?

Comment (mirkoAlic) [reply src/main/scala/io/iohk/praos/util/TransactionsGenerator.scala:36]: :)

**Review by ntallar — APPROVED** [review]
LGTM!

**Pull request merged**